File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/Address.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class Address : ApiBase
{
    [JsonProperty("address1")] public string? Address1 { get; set; }

    [JsonProperty("address2")] public string? Address2 { get; set; }

    [JsonProperty("address3")] public string? Address3 { get; set; }

    [JsonProperty("city")] public string? City { get; set; }

    [JsonProperty("zip")] public string? Zip { get; set; }

    [JsonProperty("state")] public string? State { get; set; }

    [JsonProperty("country")] public string? Country { get; set; }


    public string? ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, Address1);
        Add(sb, Address2);
        Add(sb, Address3);
        Add(sb, City);
        Add(sb, Zip);
        Add(sb, State);
        Add(sb, Country);
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/Address.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
@JsonInclude(value = JsonInclude.Include.NON_NULL)
class Address extends ApiBase implements Signable {
    private String address1;
    private String address2;
    private String address3;
    private String city;
    private String zip;
    private String state;
    private String country;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, address1);
        add(sb, address2);
        add(sb, address3);
        add(sb, city);
        add(sb, zip);
        add(sb, state);
        add(sb, country);
        deleteLast(sb);
        return sb.toString();
    }
}
